### PR TITLE
Bug/3706 replace undefined

### PR DESCRIPTION
**Pull request by eugene-manuilov**
## Summary

Addresses issue #3706

## Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [ ] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

**Comment by eugene-manuilov** [comment]
@googlebot I consent.

File: assets/js/modules/analytics-4/datastore/properties.js [hunk 1]
```diff
@@ -240,7 +240,9 @@ const baseActions = {
 	 */
 	*matchPropertyByURL( properties, url ) {
 		const registry = yield commonActions.getRegistry();
-		const urls = ( Array.isArray( url ) ? url : [ url ] ).map( normalizeURL );
+		const urls = ( Array.isArray( url ) ? url : [ url ] )
+			.filter( ( item ) => item?.length > 0 )
```
Comment (felixarntz) [inline assets/js/modules/analytics-4/datastore/properties.js:244]: Maybe use `typeof item === 'string'` instead? Things that aren't strings can still have a `length` greater than 0.

**Review by felixarntz — COMMENTED** [review]
One comment, other than that looks good. I think we should include it in 1.37.0.

File: assets/js/modules/analytics-4/datastore/properties.js (hunk 1 above)
Comment (aaemnnosttv) [reply assets/js/modules/analytics-4/datastore/properties.js:244]: We should also add a guard to `normalizeURL` which doesn't have a safe fallback if given another type. https://github.com/google/site-kit-wp/blob/371459e27494c3c2b083249c79af390a7f1139d0/assets/js/util/urls.js#L52-L54

Comment (eugene-manuilov) [reply assets/js/modules/analytics-4/datastore/properties.js]: Updated.

**Review by felixarntz — APPROVED** [review]
Great!

**Pull request merged**